### DISCUSSION TITLE
Allow default props to be set taking into account the current component props

### DISCUSSION THREAD

**Discussion by afondevilarevelli** [mantinedev/mantine#4767] [Feature requests]
Hello!

Let me show you what we wanted to do for our project so you can easily understand this feature request.

Look at this Switch Component:
![image](https://github.com/mantinedev/mantine/assets/28174678/07ca2bbb-19e4-4481-9341-457e20626291)

We wanted all Switches to look like that out of the box, so we thought using the theme object with its defaultProps for the time, BUT we cannot choose which Icon to use cause it depends whether the switch is checked or not.

It would be great we can do this easily by just doing something like:
**Switch: { defaultProps: (comp) => ({ thumbIcon: comp.checked ? Icon1 : Icon2 }) },**

Regards,
Antonio

**Comment by smilhas** [comment]
This would be great! thank you!
